### ISSUE TITLE
Volume does not bind on RC when pod dies

### ISSUE THREAD

**Issue by pmoncadaisla** [GoogleCloudPlatform/kubernetes#11974]
Hi,

I'm currently running Kubernetes on GCE.
I have a replication controller for a MySQL service, this is the YAML file:

```
kind: "ReplicationController"
apiVersion: "v1"
metadata: 
  name: "mysql-controller"
  labels: 
    state: "serving"
spec: 
  replicas: 1
  selector: 
    app: "mysql"
  template: 
    metadata: 
      labels: 
        app: "mysql"
        name: "mysql"
    spec: 
      volumes:
        - name: mysql-persistent-storage
          gcePersistentDisk:
            # This GCE PD must already exist.
            pdName: mysql-disk
            fsType: ext4
      containers:
        - resources:
            limits :
              cpu: 0.5
          image: mysql
          name: mysql
          env:
            - name: MYSQL_ROOT_PASSWORD
              # change this
              value: mysql
          ports:
            - containerPort: 3306
              name: mysql
          volumeMounts:
              # name must match the volume name below
            - name: mysql-persistent-storage
              # mount path within the container
              mountPath: /var/lib/mysql
      restartPolicy: "Always"
      dnsPolicy: "ClusterFirst"
```

The first time the pod is created by the replication controller, everything works fine, the disk is mounted on the host and binded to the container.
If the pod dies, and the replication controller attemps to create a new pod, the pod is created (and running) but "it forgets" about the volume.

Are we doing something wrong?
Is this the spected behaviour?

Thank!

**Comment by saad-ali** [comment]
Thanks for the report @pmoncadaisla 

Could you clarify what you mean by "it forgets about the volume". Do you mean that the volume is empty when the pod is recreated?

What version of Kubernetes were you running when this happened? (There was a bug in the 1.0 release that could cause disks to appear empty when a disk is rapidly detached/attached--fixed with #11561).

Are you able to repro this? If so, would you mind sharing the distilled steps?

**Comment by pmoncadaisla** [comment]
Hi,

It seems to be related to the issue #11561 you comment.

We are running kubernetes 1.0.1 (Google Container Engine).

I supposed it did not attach the volume as it was empty when pod was
recreated, yes.

To reproduce this is easy, I create the RC and when the pod is running I
execute "kubectl delete pod POD", after this a new pod will be created but
volume will be empty.

Thank you very much!

--
Enviado desde mi dispositivo móvil

**Comment by alex-mohr** [comment]
@pmoncadaisla Thanks for the report -- looks like your issue is resolved?  Please open if I misinterpreted.

**Comment by pmoncadaisla** [comment]
I think you can close it as it seems to be fixed in version 1.0.2 and we
are running 1.0.1, am I wrong?

Thanks very much!

--
Enviado desde mi dispositivo móvil

**Comment by saad-ali** [comment]
@pmoncadaisla Yes, this should be fixed in 1.0.2. If you run in to this issue with 1.0.2 please reopen this bug and don't hesitate to reach out to me. Thanks again for the report!

**Comment by pmoncadaisla** [comment]
Awesome!

I hope there will be a quick rolling-update to 1.0.2 in the Google
Container Engine!

Thanks again,

--
Enviado desde mi dispositivo móvil
